lrauv_system_tests/test/vehicle/test_battery_full_charge.cc

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/*
+ * Development of this module has been funded by the Monterey Bay Aquarium
+ * Research Institute (MBARI) and the David and Lucile Packard Foundation
+ */
+
+#include <gtest/gtest.h>
+
+#include <ignition/msgs/battery_state.pb.h>
+
+#include <chrono>
+
+#include "lrauv_system_tests/TestFixture.hh"
+
+using namespace ignition;
+
+//////////////////////////////////////////////////
+/// Test if the battery discharges with time with the specified
+/// disacharge power rate, when starting with full charge.
+TEST(BatteryTest, TestDischargeFullCharged)
+{
+  using TestFixture = lrauv_system_tests::TestFixtureWithVehicle;
+  TestFixture fixture("buoyant_tethys.sdf", "tethys");
+  uint64_t iterations = fixture.Step(100u);
+
+  transport::Node node;
+  lrauv_system_tests::Subscription<msgs::BatteryState> batterySubscription;
+  batterySubscription.Subscribe(node, "/model/tethys/battery/linear_battery/state");
+
+  fixture.Step(1000u);
+
+  auto batteryMsgs = batterySubscription.ReadMessages();
```

Comment:
@adityapande-1995 consider waiting for messages or at least asserting they are there before indexing.

Reply:
Checks length of message buffer here : https://github.com/osrf/lrauv/pull/207/commits/143a0bee08d2b44cde06720e76618f5250623783